src/main/scala/mesosphere/marathon/core/health/impl/HealthCheckActor.scala

```diff
@@ -264,13 +272,359 @@ object HealthCheckActor {
 
   // self-sent every healthCheck.intervalSeconds
   case object Tick
+
   case class GetInstanceHealth(instanceId: Instance.Id)
+
   case object GetAppHealth
+
   case class ScheduleNextHealthCheck(interval: Option[FiniteDuration] = None)
 
   case class AppHealth(health: Seq[Health])
 
   case class InstanceHealth(result: HealthResult, health: Health, newHealth: Health)
+
   case class InstancesUpdate(version: Timestamp, instances: Seq[Instance])
 
 }
+
+class MarathonHttpHealthCheckActor(
```

Comment:
Most of the helper methods copy-past code from `HealthCheckActor` and `HealthCheckWorkerActor` putting a stream around it (instead of scheduling `Tick` messages).
We could extract the common code from `HealthCheckActor` and `MarathonHttpHealthCheckActor` but this would create some kind of `HealthCheckActorBehavior` and we love those, don't we? :D